pkg/gadgets/profile/tcprtt/tracer/tracer.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !withoutebpf
+
+package tracer
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/profile/tcprtt/types"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/histogram"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -target $TARGET -type hist -cc clang tcpRTT ./bpf/tcprtt.bpf.c -- -I./bpf/ -I../../../../${TARGET}
+
+type Config struct {
+	milliseconds        bool
+	extension           bool
+	localAddrHist       bool
+	remoteAddrHist      bool
+	filterLocalAddress  uint32
+	filterRemoteAddress uint32
```

Comment:
In the new mechanism we introduced with `RunWithResult`, users shouldn't pass the configuration through the `config` but the `gadgetContext`. In fact, even if `Tracer.config` has a pre-set configuration, it will be overwritten when calling `RunWithResult` by the configuration read from the `gadgetContext`. That's why I didn't export them.

Reply:
You're right. Let's keep this unexported if you want, however I think that if we want external users consuming this gadget directly we'll have to change that as parameters rely only on strings.